src/algebra/algebra/basic.lean

```diff
@@ -593,6 +593,12 @@ def mk' (f : A →+* B) (h : ∀ (c : R) x, f (c • x) = c • f x) : A →ₐ[
 
 @[simp] lemma coe_mk' (f : A →+* B) (h : ∀ (c : R) x, f (c • x) = c • f x) : ⇑(mk' f h) = f := rfl
 
+/-- If a `ring_hom` between `R`-algebras commutes with the `algebra_map`s,
+    then it is an `alg_hom`. -/
+abbreviation of_comp_eq (f : A →+* B)
```

Comment:
Thanks for pointing this out. Yes `{ commutes' := λ r, by { rw ← h, refl }, .. f}` works and is arguably the more direct approach. Not sure about the advantage of `@[simps] def` vs `abbreviation` here though.

Reply:
`abbreviation`s are often weird because they don't generate any equation lemmas, so can end up hard to unfold (!). `@[simp] def` or `@[reducible] def` would both behave a bit better.

Note that implementing this in terms of `mk'` means that it's easier to recover the ring_hom again - in principle we should have a `mk_to_ring_hom` lemma, and that would apply here if this unfolded to `mk'` and not `mk`.

Reply:
Thanks for the explanation; I now have much better understanding of the behavior of `abbreviation` vs. `@[reducible] def`.

Just to be sure about your comment regarding `mk'` makes recovering the ring_hom easier though; is it because `mk'` has the field `to_fun := f` where `f` is the ring_hom? Indeed I checked that the (only) lemma `alg_hom.of_comp_eq_apply` generated by `@[simps]` says `⇑(of_comp_eq f h) ᾰ = f.to_fun ᾰ` when I don't provide the `to_fun` field, and says `⇑(of_comp_eq f h) ᾰ = ⇑f ᾰ` when `to_fun` is provided, which is nicer. Therefore I switched to `@[simps] def` with `to_fun` provided.

I mark this resolved but feel free to undo if there are points I don't get.

Reply:
Actually I'm gonna remove `alg_hom.of_comp_eq` because `commutes' := λ r, by { rw ← h, refl },` is easy enough, and in my use case there are already [is_localization.away.away_map.lift_eq](https://leanprover-community.github.io/mathlib_docs/ring_theory/localization/away.html#is_localization.away.away_map.lift_eq) that directly fills into the `commutes'` field and [adjoin_root.lift_hom](https://leanprover-community.github.io/mathlib_docs/ring_theory/adjoin_root.html#adjoin_root.lift_hom) which already is an alg_hom.